Documentation/check-build.sh

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# This was ported from the makefile, it can probably be converted
+# to Python (see e.g. https://stackoverflow.com/a/47657926), that
+# way we might be able to call Sphinx only once and make it quicker
+
+source_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+build_dir="${source_dir}/_build"
+warnings="${build_dir}/warnings.txt"
+spelling="${build_dir}/spelling/output.txt"
+
+target="${1:-"html"}"
+
+cd "${source_dir}"
+
+build_with_spellchecker() {
+    sphinx-build -b spelling -d "${build_dir}/doctrees" . "${build_dir}/spelling" -q -E 2>&1 > "${warnings}"
+}
+
+build_with_linkchecker() {
+    sphinx-build -b linkcheck -d "${build_dir}/doctrees" . "${build_dir}/spelling" -q -E
+}
+
+filter_warnings() {
+    grep -v \
+        -e "tabs assets" \
+        -e "misspelled words" \
+        -e "RemovedInSphinx20Warning" \
+        "${build_dir}/warnings.txt"
+}
+
+has_spelling_errors() {
+    test "$(wc -l < "${spelling}")" -ne 0 
+}
+
+describe_spelling_errors() {
+    printf "\nPlease fix the following spelling mistakes:\n"
+    # show file paths relative to repo root
+    sed 's/^/* Documentation\//' "${spelling}"
+}
+
+hint_about_wordlist_update() {
+    new_words="$(sed -E 's/^([^:]+:){2} \((.*)\)/\2/g' "${spelling}" | sort -u | tr -d '\r\n' | sed "s/'/\\\\\\\\'/g")"
+    printf "\nIf the words are not misspelled, run:\n%s %s\n" \
+        "Documentation/update-spelling_wordlist.sh" "${new_words}"
+}
+
+echo "Validating documentation (syntax, spelling)..."
+if build_with_spellchecker ; then
+    status_ok=0
+    if filter_warnings > /dev/null ;  then
+        echo "\nPlease fix the following documentation warnings:\n"
+        filter_warnings
+        status_ok=1
+    fi
+
+    if has_spelling_errors ; then
+		describe_spelling_errors
+        hint_about_wordlist_update
+		status_ok=1
+	fi
+
+    if [ "${status_ok}" -ne 0 ] ; then
+        exit 1
+    fi
+fi
+
+# TODO: Fix broken links and re-enable this
```

Comment:
I attempted fixing some of the broken links (https://github.com/cilium/cilium/pull/10576), but it looks like Sphinx' link checker will ways fail on some of our links (e.g. things it spots in our docs like `curl https://exampe.com/$(uname)`), it also seems to fail to find anchors also.

Perhaps we should just drop this, or explore ways of controlling the behaviour?

Reply:
Sure, we should keep up with links, but we some links that will keep on failing because of how this link checker works (see https://github.com/cilium/cilium/pull/10576#issuecomment-599504742).

Reply:
I think I've ran this somewhere manually in the past. The main issue with it was that is extremely slow since it's literally checking if the URL exists or not.

Reply:
Yes, and it does those checks one-by-one in a single Python thread :)